homeassistant/components/reolink/config_flow.py

```diff
@@ -0,0 +1,137 @@
+                    ): vol.In(["rtsp", "rtmp"]),
+                }
+            ),
+        )
+
+
+class ReolinkFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Reolink device."""
+
+    VERSION = 1
+
+    unique_id = None
+    host_name = None
+    port = None
+    use_https = None
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(
+        config_entry: config_entries.ConfigEntry,
+    ) -> ReolinkOptionsFlowHandler:
+        """Options callback for Reolink."""
+        return ReolinkOptionsFlowHandler(config_entry)
+
+    async def async_step_user(self, user_input=None) -> FlowResult:
+        """Handle the initial step."""
+        errors = {}
+
+        if user_input is not None:
+            try:
+                await self.async_obtain_host_settings(self.hass, user_input)
+            except CannotConnect:
+                errors[CONF_HOST] = "cannot_connect"
+            except CredentialsInvalidError:
+                errors[CONF_HOST] = "invalid_auth"
+            except ApiError:
+                errors[CONF_HOST] = "cannot_connect"
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                errors[CONF_HOST] = "unknown"
+
+            if not errors:
+                user_input[CONF_PORT] = self.port
+                user_input[CONF_USE_HTTPS] = self.use_https
+
+                await self.async_set_unique_id(self.unique_id, raise_on_progress=False)
+                self._abort_if_unique_id_configured(updates=user_input)
+
+                return self.async_create_entry(
+                    title=str(self.host_name), data=user_input
+                )
+
+        data_schema = vol.Schema(
+            {
+                vol.Required(CONF_USERNAME, default="admin"): str,
+                vol.Required(CONF_PASSWORD): str,
+                vol.Required(CONF_HOST): str,
+            }
+        )
+        if errors:
+            data_schema = data_schema.extend(
+                {
+                    vol.Optional(CONF_PORT): cv.positive_int,
+                    vol.Optional(CONF_USE_HTTPS): bool,
+                }
+            )
+
+        return self.async_show_form(
+            step_id="user", data_schema=data_schema, errors=errors
+        )
+
+    async def async_obtain_host_settings(
+        self, hass: core.HomeAssistant, user_input: dict
+    ):
+        """Initialize the Reolink host and get the host information."""
+        host = ReolinkHost(hass, user_input, {})
+
+        try:
+            if not await host.init():
+                _LOGGER.error(
+                    "Error while performing initial setup of %s:%i: failed to obtain data from device",
+                    host.api.host,
+                    host.api.port,
+                )
+                raise CannotConnect
+        except Exception as error:  # pylint: disable=broad-except
+            raise error
```

Comment:
This looks like a useless exception catch?

Reply:
yes, I have removed it, I thought it was needed to execute the "finally" block, but that is not true.